src/main/java/com/minecolonies/structures/helpers/Structure.java

```diff
@@ -222,8 +225,25 @@ public void renderStructure(@NotNull final BlockPos startingPos, @NotNull final
 
         for (final Template.BlockInfo aBlockList : blockList)
         {
-            final Block block = aBlockList.blockState.getBlock();
-            final IBlockState iblockstate = aBlockList.blockState;
+            Block block = aBlockList.blockState.getBlock();
+            IBlockState iblockstate = aBlockList.blockState;
+
+            if(block == ModBlocks.blockSubstitution)
+            {
+                continue;
+            }
+
+            if(block == ModBlocks.blockSolidSubstitution)
+            {
+                iblockstate = clientWorld.getBiome(startingPos).fillerBlock;
+                if (iblockstate.getBlock() instanceof BlockFalling)
+                {
+                    iblockstate =  Blocks.DIRT.getDefaultState();
```

Comment:
You have two spaces after the equal sign. Otherwise, looks good. one question though, is this fillerBlock, else DIRT logic done elsewhere? If so we should extract it to a private or helper method